### DISCUSSION TITLE
EA31337 v3 release dates

### DISCUSSION THREAD

**Discussion by mavidelisi** [EA31337/EA31337#395] [Releases]
Hello, kenorb. When will be EA31337 v3.? released ? Can you give me some hint/information?

Thanks

**Comment by kenorb** [comment]
Hi.
At this time, I can't give you any fixed terms.
It depends on many variables, including different challenges, time management (split with other projects), focus, bugs found in MAM trading, life distractions and many other things.
Recently one of the PC where I did some optimizations died, I'm ordering a new one soon.
Some progress can be tracked at https://github.com/orgs/EA31337/projects/1
but it depends if everything is going to be included in the first v3, or going to be pushed back to the next versions, depending on the time and effort.
And whether the next release of EA VIP is going to be published before or after (this also rely on framework dependencies and its capabilities).
Another thing to take into the consideration are few refactors, whether they can be done in the first v3 release or the next one. This includes some refactor of the existing strategies (some where already done) and introducing the new one. The current common strategies are too simple, they needs some code refreshment and it's going to take a while.
Another feature which needs to be added (either the first release or the following) are signals which returns dynamic values, so instead of strategy just returning buy or sell signal, it'll return a float value (-1.0...1.0) between strong sell and strong buy (like a weak and strong signal), this also depends on the framework code dependencies, considering that EA31337-classes v3.000.x is already feature-frozen, and at this point only bug fixes can be included.
And some automated scripts which needs some work to do EA compilation which become a quiet complex process with all indicators that are aimed to be included.
So potentially/hopefully we could see v3 at the end of the year or at the beginning (due to holidays), but based on above, it's difficult to say. I'll try to keep you updated.

**Reply by mavidelisi** [reply]
Ok. Thanks. i am waiting your publish

**Reply by kenorb** [reply]
New [v3.000 RC1](https://github.com/EA31337/EA31337/releases/tag/v3.000-rc1) has been published.

**Reply by strukturwandel** [reply]
Genereal method: "3. Clone [EA31337-classes](https://github.com/EA31337/EA31337-classes) repository into MQL4/Include folder." The old classes folder there remains in the structure so there are two classes folder (classes & EA31337-classes) or the classes folder can then be removed, because it's "replaced" by EA31337-classes? Thanks
